databricks/koalas/series.py

```diff
@@ -1452,6 +1452,146 @@ def clip(self, lower: Union[float, int] = None, upper: Union[float, int] = None)
+        ...                       [0, 1, 2, 0, 1, 2, 0, 1, 2]])
+        >>> s = ks.Series([45, 200, 1.2, 30, 250, 1.5, 320, 1, 0.3],
+        ...               index=midx)
+        >>> s
+        lama    speed      45.0
+                weight    200.0
+                length      1.2
+        cow     speed      30.0
+                weight    250.0
+                length      1.5
+        falcon  speed     320.0
+                weight      1.0
+                length      0.3
+        Name: 0, dtype: float64
+
+        >>> s.drop(labels='weight', level=1)
+        lama    speed      45.0
+                length      1.2
+        cow     speed      30.0
+                length      1.5
+        falcon  speed     320.0
+                length      0.3
+        Name: 0, dtype: float64
+
+        >>> s.drop(('lama', 'weight'))
+        lama    speed      45.0
+                length      1.2
+        cow     speed      30.0
+                weight    250.0
+                length      1.5
+        falcon  speed     320.0
+                weight      1.0
+                length      0.3
+        Name: 0, dtype: float64
+
+        >>> s.drop([('lama', 'speed'), ('falcon', 'weight')])
+        lama    weight    200.0
+                length      1.2
+        cow     speed      30.0
+                weight    250.0
+                length      1.5
+        falcon  speed     320.0
+                length      0.3
+        Name: 0, dtype: float64
+        """
+        index = labels
+        if index is not None:
+            if not isinstance(index, (str, tuple, list)):
+                raise ValueError("'index' type should be one of str, list, tuple")
+            if level is None:
+                level = 0
+            if level >= len(self._internal.index_scols):
+                raise ValueError("'level' should be less than the number of indexes")
+
+            if isinstance(index, str):
+                index = [(index,)]  # type: ignore
+            elif isinstance(index, tuple):
+                index = [index]
+            else:
+                if not (all((isinstance(rows, str) for rows in index)) or
+                        all((isinstance(rows, tuple) for rows in index))):
+                    raise ValueError("If the given index is a list, it "
+                                     "should only contains names as strings, "
+                                     "or a list of tuples that contain "
+                                     "index names as strings")
+                index = [rows if isinstance(rows, tuple) else (rows,)  # type: ignore
+                         for rows in index]
+
+            def _gen_next_lvl(level):
+                while True:
+                    yield level
+                    level += 1
+
+            drop_indexes = []
+            for rows in index:
+                lvl = _gen_next_lvl(level)
+                drop_rows = [self._internal.index_scols[next(lvl)].isin(row)
```

Comment:
`self._internal.index_scols[next(lvl)] == row`?

Reply:
@ueshin , Thanks for review :)

I think maybe there are some confusing since my improper naming. 😿 

`self._internal.index_scols[next(lvl)]` is a specific index column of internal spark dataframe like: `__index_level_0__`,

and `row` is a specific index value from user parameter like `'A'` or `'B'`, etc.

i think the variable name `row` might be better change to `idx`.

Reply:
i intended that `row` as an internal spark dataframe's row,
but i think it was improper naming.

Reply:
ah, sorry that I didn't make clear.
I just thought the `row` is a single value, so we don't need to use `isin()` here.

Reply:
@ueshin oh i got it. maybe it was my misunderstood.
Anyway i think still maybe `idx` is better than `row` with but no 100% convinced of it though.